include/envoy/network/connection.h

```diff
@@ -151,15 +151,15 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   /**
    * @return The address of the remote client.
    */
-  virtual const Address::Instance& remoteAddress() const PURE;
+  virtual Network::Address::InstanceConstSharedPtr remoteAddress() const PURE;
```

Comment:
I think these should be documented that they will never return nullptr (unless you plan to change that)